namesilo.php

```diff
@@ -2929,21 +2963,37 @@ protected function getPrices()
         );
 
         if ($cache) {
-            return unserialize(base64_decode($cache));
+            $result = unserialize(base64_decode($cache));
         }
 
         Loader::loadModels($this, ['Currencies']);
 
-        $row = $this->getRow();
+        if (!$result) {
+            $row = $this->getRow();
+            $api = $this->getApi(
+                $row->meta->user,
+                $row->meta->key,
+                $row->meta->sandbox == 'true'
+            );
+            $result = $api->submit('getPrices')->response();
 
-        $api = $this->getApi(
-            $row->meta->user,
-            $row->meta->key,
-            $row->meta->sandbox == 'true'
-        );
-        $result = $api->submit('getPrices')->response();
-        $tlds = [];
+            // Save the TLDs results to the cache
```

Comment:
Below we have a condition on (isset($result->detail) && $result->detail == 'success').  I think caching should only happen if this check is successful.